File: types/mime.cpp
#include <filesystem>
#include "mime.h"

#include "types/String.h"

bool quark::MTMime::isImage(const std::string& path)
{
    const auto stdPath = std::filesystem::path(path);
    auto ext = stdPath.extension().string();
    ext = PSString::toLower(ext);
    if (ext == ".jpg" || ext == ".jpeg" || ext == ".png" || ext == ".gif" || ext == ".webp")
        return true;
    return false;
}


std::string quark::MTMime::getMimeType(const std::string& path)
{
    const auto stdPath = std::filesystem::path(path);
    auto ext = stdPath.extension().string();
    ext = PSString::toLower(ext);

    if (ext == ".txt")
        return "text/plain";
    if (ext == ".htm" || ext == ".html")
        return "text/html";
    if (ext == ".js")
        return "application/javascript";
    if (ext == ".css")
        return "text/css";
    if (ext == ".jpg" || ext == ".jpeg")
        return "image/jpeg";
    if (ext == ".png")
        return "image/png";
    if (ext == ".json")
        return "application/json";
    if (ext == ".gif")
        return "image/gif";
    if (ext == ".webp")
        return "application/webp";

    return "application/octet-stream";
}

File: atom/common/utils/mime.ts

export function getMimeType(path: string): string {
    const mimeMap: Map<string, string> = new Map<string, string>()
    mimeMap.set('txt', 'text/plain')
    mimeMap.set('html', 'text/html')
    mimeMap.set('css', 'text/css')
    mimeMap.set('js', 'text/javascript')
    mimeMap.set('json', 'application/json')
    mimeMap.set('xml', 'application/xml')

    const ext = path.split('.').pop()
    if (!ext) {
        return 'text/plain'
    }
    return mimeMap.get(ext) || 'text/plain'
}

export type getType = typeof getMimeType

export function isImageType(type: string): boolean {
    const lowerType = type.toLowerCase().trim()
    return lowerType.startsWith('image/') ||
        lowerType.endsWith(".jpg") ||
        lowerType.endsWith(".jpeg") ||
        lowerType.endsWith(".png") || lowerType.endsWith(".gif")
}
